diff --git a/dist/metro/index.cjs b/dist/metro/index.cjs
--- a/dist/metro/index.cjs
+++ b/dist/metro/index.cjs
@@ -160,7 +160,10 @@ const webResolver = ({
   if (filename.startsWith("createOrderedCSSStyleSheet.")) {
     return resolver(context, `uniwind/components/createOrderedCSSStyleSheet`, platform);
   }
-  if (!isIndex || module === void 0 || !SUPPORTED_COMPONENTS.includes(module) || context.originModulePath.endsWith(`${module}${node_path.sep}index.js`)) {
+  const originNodeModulesPath = context.originModulePath.split(`${node_path.sep}node_modules${node_path.sep}`).at(-1) ?? "";
+  const isFromReactNativeWeb = originNodeModulesPath !== context.originModulePath && originNodeModulesPath.startsWith(`react-native-web${node_path.sep}`);
+  const isReactNativeAnimated = context.originModulePath.includes(`${node_path.sep}Animated${node_path.sep}components${node_path.sep}`);
+  if (isFromReactNativeWeb && !isReactNativeAnimated || !isIndex || module === void 0 || !SUPPORTED_COMPONENTS.includes(module)) {
     return resolution;
   }
   return resolver(context, `uniwind/components/${module}`, platform);
diff --git a/dist/metro/index.mjs b/dist/metro/index.mjs
--- a/dist/metro/index.mjs
+++ b/dist/metro/index.mjs
@@ -152,7 +152,10 @@ const webResolver = ({
   if (filename.startsWith("createOrderedCSSStyleSheet.")) {
     return resolver(context, `uniwind/components/createOrderedCSSStyleSheet`, platform);
   }
-  if (!isIndex || module === void 0 || !SUPPORTED_COMPONENTS.includes(module) || context.originModulePath.endsWith(`${module}${sep}index.js`)) {
+  const originNodeModulesPath = context.originModulePath.split(`${sep}node_modules${sep}`).at(-1) ?? "";
+  const isFromReactNativeWeb = originNodeModulesPath !== context.originModulePath && originNodeModulesPath.startsWith(`react-native-web${sep}`);
+  const isReactNativeAnimated = context.originModulePath.includes(`${sep}Animated${sep}components${sep}`);
+  if (isFromReactNativeWeb && !isReactNativeAnimated || !isIndex || module === void 0 || !SUPPORTED_COMPONENTS.includes(module)) {
     return resolution;
   }
   return resolver(context, `uniwind/components/${module}`, platform);
